streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -199,30 +194,33 @@ void time(final Time time) {
      */
     @Override
     public void configure(Map<String, ?> configs) {
-        numStandbyReplicas = (Integer) configs.get(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG);
+        final StreamsConfig streamsConfig = new StreamsConfig(configs);
 
         // Setting the logger with the passed in client thread name
-        logPrefix = String.format("stream-thread [%s] ", configs.get(CommonClientConfigs.CLIENT_ID_CONFIG));
+        this.logPrefix = String.format("stream-thread [%s] ", streamsConfig.getString(CommonClientConfigs.CLIENT_ID_CONFIG));
         final LogContext logContext = new LogContext(logPrefix);
         this.log = logContext.logger(getClass());
 
-        Object o = configs.get(StreamsConfig.InternalConfig.STREAM_THREAD_INSTANCE);
+        Object o = configs.get(StreamsConfig.InternalConfig.TASK_MANAGER_FOR_PARTITION_ASSIGNOR);
         if (o == null) {
-            KafkaException ex = new KafkaException("StreamThread is not specified");
+            KafkaException ex = new KafkaException("TaskManager is not specified");
             log.error(ex.getMessage(), ex);
             throw ex;
         }
 
-        if (!(o instanceof ThreadDataProvider)) {
-            KafkaException ex = new KafkaException(String.format("%s is not an instance of %s", o.getClass().getName(), ThreadDataProvider.class.getName()));
+        if (!(o instanceof TaskManager)) {
```

Comment:
Wondering why we don't keep the interface and use TaskManager directly?

Reply:
I once tried to still maintain an interface but then I realized most of the public functions are needed in the assignor after we moved most of its state into the TaskManager:

`builder(), prevActiveTaskIds(), cachedTaskIds(), processorId()` and a bunch of setters and updators. The interface num.functions are actually more than what we have today, so I decided to have the dependency on that class directly. WDYT?